flakeModule.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, config, system, ... }:
+let
+  inherit (lib) mkOption types;
+
+  nixpkgsOverlaySubmodule = types.mkOptionType {
+    name = "nixpkgsOverlay";
+    description = "A nixpkgs overlay function";
+    descriptionClass = "noun";
+    check = lib.isFunction;
+    merge = _loc: defs:
+      let
+        logWarning =
+          if builtins.length defs > 1
+          then builtins.trace "WARNING[nixpkgs.flakeModule]: Multiple overlays are applied in arbitrary order." null
+          else null;
+        overlays =
+          map (x: x.value)
+            (builtins.seq
+              logWarning
+              defs);
+      in
+      lib.composeManyExtensions overlays;
+  };
+
+  nixpkgsSubmodule = with types; submodule {
+    options = {
+      overlays = mkOption {
+        type = listOf nixpkgsOverlaySubmodule;
+        description = "Nixpkgs overlays";
```

Comment:
A link to the Nixpkgs manual would be helpful.

Reply:
> Hi, and thank you for proposing a contribution!
> 
> This is quite a significant addition, so I've made thorough first review as you can see.
> 
> I think your commit messages are really interesting. Do you have tooling to process those sexprs? We do have some guidelines for the format of commit messages. Would you be willing to adapt to those? They're listed [here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes). They're probably not hugely specific rules for these commits, but I know that the other contributors will appreciate commit messages that fit in with the rest; certainly for the title part of the message.
> 
> I'd also like to ask you to squash your commits in the end, but keep a commit boundary between changes that could be used separately, such as for example the addition of `types.overlay`.

Yes, those are a proof of concept lang being developped dynamically. They will be processed eventually, when the first sajban implementation is out. For sure, I will learn about NixOS style commits and squash this PR when done. This is just a documenting PR. Thank you so much for flake-parts and for your review. I will be sending you some ouf our tokens when they come out in a few weeks. I'd be happy to have you on a live podcast to discuss Nix and its future. Cheers.